storage/flux/table.gen.go

```diff
@@ -97,6 +97,76 @@ func (t *floatTable) advance() bool {
 	return true
 }
 
+// window table
+type floatWindowTable struct {
+	floatTable
+	windowEvery int64
+	arr         *cursors.FloatArray
+	idxInArr    int
+}
+
+func newFloatWindowTable(
+	done chan struct{},
+	cur cursors.FloatArrayCursor,
+	bounds execute.Bounds,
+	every int64,
+	key flux.GroupKey,
+	cols []flux.ColMeta,
+	tags models.Tags,
+	defs [][]byte,
+	cache *tagsCache,
+	alloc *memory.Allocator,
+) *floatWindowTable {
+	t := &floatWindowTable{
+		floatTable: floatTable{
+			table: newTable(done, bounds, key, cols, defs, cache, alloc),
+			cur:   cur,
+		},
+		windowEvery: every,
+	}
+	t.readTags(tags)
+	t.advance()
+
+	return t
+}
+
+func (t *floatWindowTable) advance() bool {
+	if t.arr == nil {
+		t.arr = t.cur.Next()
+		if t.arr.Len() == 0 {
+			t.arr = nil
+			return false
+		}
+		t.idxInArr = 0
+	}
+
+	// Retrieve the buffer for the data to avoid allocating
+	// additional slices. If the buffer is still being used
+	// because the references were retained, then we will
+	// allocate a new buffer.
+	columnReader := t.allocateBuffer(1)
```

Comment:
We talked offline about this loop and ways we can optimize it further in the future for `aggregateWindow()`. It looks good for its current purpose so I don't see anything else I would change.